### DISCUSSION TITLE
Same network for VMs as KVM hosts?

### DISCUSSION THREAD

**Discussion by dR3b** [apache/cloudstack#8998] [Q&A] [answered]
Hi, I would like to set up a Apache Cloudstack cluster for testing purposes. I have one network (no VLAN) and 4 Intel NUCs available.

```
  ---> ROUTER ---> NETWORK1 ---> NUC1 == Cloudstack Management, NFS (PS and SS)
                            ---> NUC2 == KVM Host 1, Cloudstack Agent
                            ---> NUC3 == KVM Host 2, Cloudstack Agent
                            ---> NUC4 == KVM Host 3, Cloudstack Agent
```
```
Network:      10.20.30.0/24
Gateway:      10.20.30.1
Subnet:       255.255.255.0
DNS:          10.20.30.1
DNS2:         8.8.4.4
DHCP Range:   10.20.30.50-150
```

My question: What settings are necessary so that I can create the VMs in the same subnet as the NUCs (DHCP 10.20.30.200-240)?

Thank you very much for your help.

**Comment by weizhouapache** [comment]
I think a shared network with vlan=untagged is what you need.
You can use an advanced zone or advanced zone with security groups, or even a basic zone.

**Comment by dR3b** [comment]
@weizhouapache 

Thanks! After several pages in the documentation[1], I was able to complete the installation.

- [x] System-VM consoleproxy, Up and running
- [x] System-VM secondarystoragevm, Up and running
- [x] Virtual-Router, defaultGuestNetwork, Up and running
- [x] Compute-Instance, arch1-1, Running (192.168.1.196)

But now I have another question: I cannot ping the IP of “arch1-1” via a PC in the same network (192.168.1.53). From my Cloudstack Management/KVM host, however, it's possible. What have I forgotten?

[1] https://docs.cloudstack.apache.org/en/latest/adminguide/networking_and_traffic.html#configuring-a-shared-guest-network

**Reply by weizhouapache** [reply]
@dR3b 
Is IP forwarding enabled?
The following command should return 1

cat /proc/sys/net/ipv4/ip_forward

**Reply by dR3b** [reply]
@weizhouapache Yes!
```
root@nuc1:/# cat /proc/sys/net/ipv4/ip_forward
1
```

**Comment by weizhouapache** [comment, marked as answer]
@dR3b 
a user has replied to to this thread in the users mailing list, no idea why it is not present here

![image](https://github.com/apache/cloudstack/assets/57355700/cc9f9b3e-09e9-47f7-ba23-1b8b0667872a)

**Comment by dR3b** [comment]
@weizhouapache Thanks! This works, but deletes all the IPtable rules on the host. I have not created any rules myself. These rules must be from Cloudstack or UFW.

**Reply by dR3b** [reply]
@weizhouapache 
```
root@nuc1:/# apt install cloudstack-agent
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  ufw
Suggested packages:
  rsyslog
The following NEW packages will be installed:
  cloudstack-agent ufw
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/90.3 MB of archives.
After this operation, 105 MB of additional disk space will be used.
Do you want to continue? [Y/n]
```
